### ISSUE TITLE
Causing several bugs when empty headers

### ISSUE THREAD

**Issue by codpot** [awslabs/aws-serverless-express#462]
Due to Node.js `http.IncomingMessage` message.headers is Object, Express.js also returning {} when empty http headers.

https://nodejs.org/api/http.html#http_message_headers

https://github.com/expressjs/express/blob/06d11755c99fe4c1cddf8b889a687448b568472d/lib/request.js#L38

https://github.com/expressjs/express/blob/06d11755c99fe4c1cddf8b889a687448b568472d/lib/request.js#L82

```js
const express = require('express');
const app = express();

app.use((req, res, next) => {
  console.log(req.headers);
  next();
});

app.get('/', (req, res) => {
  res.send('Hello World!');
});

app.listen(5000);
```

```bash
curl -v http://localhost:5000/ -H 'User-Agent:' -H 'Accept:' -H 'Host:'
```
![image](https://user-images.githubusercontent.com/5399876/134478599-baa59c65-d35e-4765-8fd2-f247f567a130.png)

![image](https://user-images.githubusercontent.com/5399876/134478462-8bf7c9f3-99b6-4c4a-ba2a-e295eef59349.png)

So that, several express based libraries treat Express.js `req.headers` as Nonnullable Object. 

Following line cause problem.

https://github.com/vendia/serverless-express/blob/6adac653fe1daf1dd44ff6aa17dbf08f07662173/src/event-sources/utils.js#L36

`AWS Console -> API Gateway -> APIs -> Resources -> Test` with empty Headers causing problems.

```
2021-09-23T08:52:34.010Z	08748566-1f43-409d-9f2b-0b1fc9cf805a	ERROR	TypeError: Cannot read property 'origin' of null
    at /var/task/node_modules/cors/lib/index.js:219:40
    at optionsCallback (/var/task/node_modules/cors/lib/index.js:199:9)
    at corsMiddleware (/var/task/node_modules/cors/lib/index.js:204:7)
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/task/node_modules/express/lib/router/index.js:317:13)
    at /var/task/node_modules/express/lib/router/index.js:284:7
    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:335:12)
    at next (/var/task/node_modules/express/lib/router/index.js:275:10)
    at /var/task/dist/src/lambda/main.js:18:9
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
```

```
2021-09-23T08:07:35.502Z	689819f8-4baf-4a91-9a0a-9206c111b24c	ERROR	TypeError: Cannot read property 'if-modified-since' of null
    at fresh (/var/task/node_modules/fresh/index.js:35:33)
    at IncomingMessage.fresh (/var/task/node_modules/express/lib/request.js:477:12)
    at ServerResponse.send (/var/task/node_modules/express/lib/response.js:206:11)
    at /var/task/node_modules/@nestjs/swagger/dist/swagger-module.js:37:54
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at next (/var/task/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/var/task/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/var/task/node_modules/express/lib/router/layer.js:95:5)
    at /var/task/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/var/task/node_modules/express/lib/router/index.js:335:12)
```

Also, tested with POST method with empty headers and non-empty body causing problem in serverless-express.

https://github.com/vendia/serverless-express/blob/6adac653fe1daf1dd44ff6aa17dbf08f07662173/src/event-sources/utils.js#L44

```
2021-09-23T08:54:32.228Z	787d03d9-ec52-49cf-96ea-762d875bea70	ERROR	Unhandled Promise Rejection 	
{
    "errorType": "Runtime.UnhandledPromiseRejection",
    "errorMessage": "TypeError: Cannot set property 'content-length' of null",
    "reason": {
        "errorType": "TypeError",
        "errorMessage": "Cannot set property 'content-length' of null",
        "stack": [
            "TypeError: Cannot set property 'content-length' of null",
            "    at getRequestValuesFromEvent (/var/task/node_modules/@vendia/serverless-express/src/event-sources/utils.js:44:31)",
            "    at Object.getRequestValuesFromApiGatewayEvent [as getRequest] (/var/task/node_modules/@vendia/serverless-express/src/event-sources/aws/api-gateway-v1.js:3:60)",
            "    at forwardRequestToNodeServer (/var/task/node_modules/@vendia/serverless-express/src/transport.js:133:37)",
            "    at /var/task/node_modules/@vendia/serverless-express/src/configure.js:71:9",
            "    at new Promise (<anonymous>)",
            "    at proxy (/var/task/node_modules/@vendia/serverless-express/src/configure.js:58:12)",
            "    at handler (/var/task/node_modules/@vendia/serverless-express/src/configure.js:95:12)",
            "    at Runtime.handler (/var/task/dist/src/lambda/main.js:53:12)",
            "    at Runtime.handleOnce (/var/runtime/Runtime.js:66:25)"
        ]
    },
    "promise": {},
    "stack": [
        "Runtime.UnhandledPromiseRejection: TypeError: Cannot set property 'content-length' of null",
        "    at process.<anonymous> (/var/runtime/index.js:35:15)",
        "    at process.emit (events.js:400:28)",
        "    at process.emit (domain.js:470:12)",
        "    at processPromiseRejections (internal/process/promises.js:245:33)",
        "    at processTicksAndRejections (internal/process/task_queues.js:96:32)"
    ]
}
```

TL;DR
When `event.multiValueHeaders` and `event.headers` are null, `headers` need to return `{}` not `null`.
